packages/magic/src/index.ts

```diff
@@ -0,0 +1,206 @@
+import type { WalletInit, APIKey, EIP1193Provider } from '@web3-onboard/common'
+
+function magic(options: APIKey): WalletInit {
+  const { apiKey } = options
+  const walletName = 'Magic Wallet'
+
+  return () => {
+    return {
+      label: walletName,
+      getIcon: async () => (await import('./icon.js')).default,
+      getInterface: async ({ EventEmitter, BigNumber, chains }) => {
+        const { Magic } = await import('magic-sdk')
+        const loginModal = (await import('./login-modal.js')).default
+        const brandingHTML = (await import('./branding.js')).default
+        let loggedIn: boolean = false
+
+        const {
+          createEIP1193Provider,
+          ProviderRpcErrorCode,
+          ProviderRpcError
+        } = await import('@web3-onboard/common')
+
+        const emitter = new EventEmitter()
+
+        let currentChain = chains[0].id
+
+        let customNodeOptions = {
+          chainId: parseInt(chains[0].id, 10),
+          rpcUrl: chains[0].rpcUrl
+        }
+
+        let magicInstance = new Magic(apiKey, {
+          network: customNodeOptions
+        })
+
+        const loginWithEmail = async (emailAddress: string) => {
+          try {
+            await magicInstance.auth.loginWithMagicLink({ email: emailAddress })
+            return await magicInstance.user.isLoggedIn()
+          } catch (err) {
+            throw new Error(
+              `$An error occured while connecting your Magic wallet, please try again: ${err}`
+            )
+          }
+        }
+
+        const handleLogin = async () => {
+          loggedIn = await loginModal({
+            walletName: walletName,
+            brandingHTMLString: brandingHTML,
+            emailLoginFunction: loginWithEmail
+          })
+        }
+
+        if (!loggedIn) await handleLogin()
```

Comment:
Does this call to login need to be here? The `core` module will call the `eth_requestAccounts` once it gets the provider, so i think that `handleLogin` might be called twice in quick succession.